app/developers/deliver_routes.py

```diff
@@ -921,86 +877,45 @@ def collection_confirmation(submission_id: UUID) -> ResponseReturnValue:
-    # todo: questions with multiple inputs will need to think this through a bit more
-    form = build_question_form(question, expression_context=expression_context)(data=expression_context)
+    if runner.tasklist_form.validate_on_submit():
+        with suppress(ValueError):
+            runner.submit(interfaces.user.get_current_user())
 
-    if not submission_helper.is_question_visible(question, submission_helper.expression_context):
-        current_app.logger.warning(
-            "Routed to a question that is not visible for submission_id=%(submission_id)s, "
-            "question_id=%(question_id)s, is_submitted=%(is_submitted)s",
-            dict(
-                submission_id=str(submission_helper.id),
-                question_id=str(question.id),
-                is_submitted=form.is_submitted(),
-            ),
-        )
-        return redirect(
-            url_for("developers.deliver.check_your_answers", submission_id=submission_id, form_id=question.form_id)
-        )
+            # todo: for now we'll email and confirm on submission but later DGF form builder will likely
+            #       just return to the tasklist editing page on completion
+            notification_service.send_collection_submission(runner.submission.submission)
+            return redirect(url_for("developers.deliver.collection_confirmation", submission_id=runner.submission.id))
 
-    if submission_helper.is_completed:
-        if form.is_submitted():
-            # TODO: Add an error flash message?
-            pass
-        return redirect(
-            url_for("developers.deliver.check_your_answers", submission_id=submission_id, form_id=question.form_id)
-        )
+    return render_template(
+        "developers/deliver/collection_tasklist.html",
+        runner=runner,
+    )
 
-    if form.validate_on_submit():
-        submission_helper.submit_answer_for_question(question.id, form)
 
-        if request.args.get("source") == FormRunnerSourceEnum.CHECK_YOUR_ANSWERS:
-            return redirect(
-                url_for("developers.deliver.check_your_answers", submission_id=submission_id, form_id=question.form_id)
-            )
+@developers_deliver_blueprint.route("/submissions/<uuid:submission_id>/<uuid:question_id>", methods=["GET", "POST"])
+@is_platform_admin
+@auto_commit_after_request
+def ask_a_question(submission_id: UUID, question_id: UUID) -> ResponseReturnValue:
+    source = request.args.get("source")
+    runner = DGFFormRunner.load(
+        submission_id=submission_id, question_id=question_id, source=FormRunnerState(source) if source else None
+    )
 
-        next_question = submission_helper.get_next_question(current_question_id=question_id)
-        if next_question:
-            return redirect(
-                url_for("developers.deliver.ask_a_question", submission_id=submission_id, question_id=next_question.id)
-            )
+    if not runner.validate_can_show_question_page():
```

Comment:
nit for future thought: should instantiating the runner run this check instead? Could be a bit magical, could be a bit reliable? https://github.com/communitiesuk/funding-service/commit/cd054ba6c2b07b04b71b84bab7d676d4bc13be6b